### DISCUSSION TITLE
[@types/archiver] broken by `glob@9`

### DISCUSSION THREAD

**Discussion by alecgibson** [DefinitelyTyped/DefinitelyTyped#64529] [Issues with a @types package] [answered]
[`glob@9`](https://github.com/isaacs/node-glob/blob/main/changelog.md#90) has been released, which — surprisingly — breaks these type definitions.

This is surprising, since `archiver` doesn't depend on `glob`; it depends on [`readdir-glob`](https://github.com/archiverjs/node-archiver/blob/b5cc14cc97cc64bdca32c0cbe9d660b5b979be7c/package.json#L35), but `@types/archiver` uses the [types from `glob`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/5f13a942cd2adde3e996ba2f5348eb2f7aa98c99/types/archiver/index.d.ts#L11), which now breaks with:

```
node_modules/@types/archiver/index.d.ts:88:41 - error TS2503: Cannot find namespace 'glob'.

88         glob(pattern: string, options?: glob.IOptions, data?: Partial<EntryData>): this;
                                           ~~~~


Found 1 error in node_modules/@types/archiver/index.d.ts:88

```

[a comment by an automated account is not shown]

**Comment by alecgibson** [comment, marked as answer]
I assume the correct solution is to write type definitions for `@types/readdir-glob` and use those?

**Reply by dolanmiu** [reply]
Good idea.

i have just created type definitions for `readdir-glob`:

https://github.com/DefinitelyTyped/DefinitelyTyped/pull/64532

**Reply by dolanmiu** [reply]
After that, we can update `archiver`

**Reply by dolanmiu** [reply]
Archiver types PR is here:

https://github.com/DefinitelyTyped/DefinitelyTyped/pull/64611

**Reply by dolanmiu** [reply]
All merged, should be fixed
